source/Calamari/Kubernetes/KubernetesContextScriptWrapper.cs

```diff
@@ -421,6 +425,59 @@ void SetupContextForAmazonServiceAccount(string @namespace, string clusterUrl, s
                 var clusterName = variables.Get(SpecialVariables.EksClusterName);
                 log.Info($"Creating kubectl context to {clusterUrl} (namespace {@namespace}) using EKS cluster name {clusterName}");
 
+                if (!TrySetAws())
+                {
+                    log.Warn("Could not find the aws cli, falling back to the aws-iam-authenticator.");
+                }
+                else
+                {
+                    try
+                    {
+                        var region = Regex.Match(clusterUrl, @"(us(-gov)?|ap|ca|cn|eu|sa)-(central|(north|south)?(east|west)?)-\d", RegexOptions.IgnoreCase).Value;
+                        if (string.IsNullOrWhiteSpace(region))
+                        {
+                            log.Error("The EKS cluster Url specified should contain a valid aws region name");
+                        }
+                        
+                        var awsCliCommandRes = ExecuteCommandAndReturnOutput(aws, "--version").FirstOrDefault();
+                        var awsCliVersion = Regex.Match(awsCliCommandRes, @"aws-cli\/(\d+\.)?(\d+\.)?(\*|\d+)").Value.Replace("aws-cli/", string.Empty);
+                        var awsCliSemVer = new SemanticVersion(awsCliVersion);
+                        var eksGetTokenVersion = new SemanticVersion("1.16.156");
+                        var validAwsVersion = awsCliSemVer.CompareTo(eksGetTokenVersion) == 1;
```

Comment:
The documentation only specifies the meaning when the result of `CompareTo` is greater than zero, less than zero and equal to zero. The meaning of exact number is unspecified. I suggest using `> 0` over `== 1`.